frontend_app_thread/src/container/Thread.jsx

```diff
@@ -285,7 +288,25 @@ export class Thread extends React.Component {
     props.appContentChangeComment(e, state.content, this.setState.bind(this), state.appName)
   }
 
+  searchForMentionInQuery = async (query) => {
+    return await this.props.searchForMentionInQuery(query, this.state.content.workspace_id)
+  }
+
   handleClickValidateNewCommentBtn = async () => {
+    const { state } = this
+    const knownMentions = state.config.workspace.memberList.map(member => `@${member.username}`)
+    const comment = state.timelineWysiwyg ? tinymce.activeEditor.getContent() : state.newComment
+    const invalidMentionList = getInvalidMentionsList(comment, knownMentions)
+
+    if (invalidMentionList.length > 0) {
+      this.setState({
+        invalidMentionList: invalidMentionList,
+        showInvalidMentionPopupInComment: true
+      })
+    } else this.handleClickValidateAnywayNewComment()
+  }
+
+  handleClickValidateAnywayNewComment = () => {
```

Comment:
Maybe put this duplicated code in frontend lib?

Reply:
Changed.